### PR TITLE
feat: Add aws:glue-dev-endpoint resource and delete action

### PR DESCRIPTION
**Pull request by jtroberts83**
With public dev endpoints in glue being a security issue this will allow for finding and deleting the public endpoints

### CODE REVIEW

**Comment by jtroberts83** [comment]
@kapilt I have implemented the w.submit pieces and got it working.  Hopefully it looks good now.

File: c7n/resources/glue.py [hunk 1]
```diff
@@ -80,3 +81,57 @@ def delete_connection(self, r):
...
+            actions:
+              - type: delete
+    """
+    schema = type_schema('delete')
+    permissions = ('glue:DeleteDevEndpoint',)
+
+    def delete_dev_endpoint(self, client, endpoint_set):
+        try:
+            for e in endpoint_set:
+                client.delete_dev_endpoint(EndpointName=e['EndpointName'])
```
Comment (kapilt) [inline c7n/resources/glue.py]: the try except should be around the inner loop, else a resumable error on one would stop the rest of the set.

Comment (jtroberts83) [reply c7n/resources/glue.py]: Ah crap, I will get that corrected

**Comment by jtroberts83** [comment]
@kapilt  Ready for merge

**Comment by jtroberts83** [comment]
Thanks

**Pull request merged**